files/src/main/java/com/afollestad/materialdialogs/files/DialogFileChooserExt.kt

```diff
@@ -62,7 +63,8 @@ fun MaterialDialog.selectedFile(): File? {
  */
 @SuppressLint("CheckResult")
 fun MaterialDialog.fileChooser(
-  initialDirectory: File = getExternalStorageDirectory(),
+  context: Context,
+  initialDirectory: File? = context.getExternalFilesDir(),
```

Comment:
Have you verified this works on older Android versions?

Reply:
Following with android documentation related `getExternalStoragePublicDirectory(String type)`

> This method was deprecated in API level 29. To improve user privacy, direct access to shared/external storage devices is deprecated. When an app targets Build.VERSION_CODES.Q, the path returned from this method is no longer directly accessible to apps. Apps can continue to access content stored on shared/external storage by migrating to alternatives such as Context#getExternalFilesDir(String), MediaStore, or Intent#ACTION_OPEN_DOCUMENT.

I decided to migrate it to recommended `getExternalFilesDir(String)` https://developer.android.com/reference/android/content/Context.html#getExternalFilesDir(java.lang.String). This method is added in API level 8. 

The difference  main difference between them are the following: 
- `getExternalFilesDir()`
It returns the path to files folder inside Android/data/data/your_package/ on your SD card. It is used to store any required files for your app (e.g. images downloaded from web or cache files). Once the app is uninstalled, any data stored in this folder is gone too.

- `getExternalStorageDirectory()` (Deprecated)
It returns the root path to your SD card (e.g mnt/sdcard/). If you save data on this path and uninstall the app, that data won't be lost.